src/common/helpers/data-extract/data-extract.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @typedef  { 'name' | 'file' | 'text' |' radio' | 'address' | 'checkbox' } AnswerType
+ * @typedef  {{ type: AnswerType, value: string, displayText: string }} AnswerData
+ * @typedef  {{ question: string, questionKey: string, answer: AnswerData }} QuestionAnswerData
+ * @typedef  {{ title: string, sectionKey: string, questionAnswers: QuestionAnswerData[]}} SectionData
+ */
+
+/**
+ * @param {string} questionKey
+ * @param {string} sectionKey
+ * @param {SectionData[]} sections
+ * @returns {QuestionAnswerData|undefined}
+ */
+export const getQuestionFromSections = (questionKey, sectionKey, sections) => {
+  const section = Array.isArray(sections)
```

Comment:
I don't think we need this `Array.isArray` checks here, because the types specify this:

```javascript
const section = sections.find((section) => section.sectionKey === sectionKey)
const question = section?.questionAnswers.find((question) => question.questionKey === questionKey)

return question
```

Looks like it'd be enough?

Reply:
But the types are not checked at runtime so a call could send something that is not an array?

Reply:
@joseluisgraa  - yup - we will handle that case in the Joi validation ticket, though

Reply:
done